src/Calendar.elm

```diff
@@ -133,15 +134,16 @@ view state =
             ]
 
 
-dayRangeOfWeek date =
+weekRangeFromDate : Date -> List Date
+weekRangeFromDate date =
     let
-        firstOfWeek =
-            Date.Extra.floor Date.Extra.Week date
```

Comment:
Date.Extra uses Monday as the starting day of the week, so using `floor` or `ceiling` with `Date.Extra.Week` is synonymous to using it with Monday (which isn't what you want here).